.github/workflows/check_need_autoupgrade_pr_label.yml

```diff
@@ -0,0 +1,81 @@
+# This workflow allows you to check potential changes to the structure of the database,
+# and to apply the “Needs autoupgrade PR” label if this is the case
+name: Check the label 'Needs autoupgrade PR'
+
+on: pull_request_target
+
+jobs:
+  install_prestashop_and_dump:
+    name: Install Prestashop and create database dump
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        branch: [ base, head ]
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.branch == 'base' && github.event.pull_request.base.ref || github.event.pull_request.head.ref }}
```

Comment:
It took me some time to understand how this line was defined.
Would this work ?

```suggestion
          ref: ${{ format('github.event.pull_request.{0}.ref', matrix.branch) }}
```

https://docs.github.com/en/actions/learn-github-actions/expressions#format

Reply:
I think so. I understand that the homemade ternary is strange, but in terms of understanding, the format function isn't better for me if you don't known it. I'll try to add it because it's sexier anyway :+1:

Reply:
To myself: ![At least you tried](https://i.redd.it/wmrx1wbtpjg01.jpg)